src/values/NativeFunctionValue.js

```diff
@@ -71,9 +71,20 @@ export default class NativeFunctionValue extends FunctionValue {
     });
 
     if (name) {
-      this.name = name instanceof SymbolValue ? `[${name.$Description || "native"}]` : name;
+      if (name instanceof SymbolValue && name.$Description instanceof AbstractValue) {
+        this.name = name.$Description;
+      } else {
+        if (name instanceof SymbolValue) {
+          invariant(typeof name.$Description === "string");
```

Comment:
Does name.$Description have to be of type string, couldn't it also be undefined?